### PR TITLE
Higher level test

### PR DESCRIPTION
**Pull request by karen-olson**
Additions in this PR: 

- Edit ReadMe to reflect the current state of the project
- Create an acceptance test for the echo server and add it to the GitHub Actions CI workflow

### CODE REVIEW

File: bin/acceptance-test [hunk 1]
```diff
@@ -1,24 +1,77 @@
...
+  echo $!
+}
+
+function wait_for_server {
+  local host=$1
+  local port=$2
+
+  echo "Waiting for server to start"
+  local max_attempts=5
+  echo "$max_attempts"
```
Comment (damonkelley) [inline bin/acceptance-test:16]: **question** There are quite a few new `echo` statements. Are all of these valuable?

```bash
echo $max_attempts
# ...

echo "Entered the loop"

# ...

echo "connected"

# ...
```

File: bin/acceptance-test [hunk 2]
```diff
@@ -1,24 +1,77 @@
...
+    echo -n "."
+  done
+}
+
+function connect_with_netcat {
+  local os=uname
+  local host=$1
+  local port=$2
+  local input=$3
+  local timeout_seconds=0
```
Comment (damonkelley) [inline bin/acceptance-test:37]: **praise** I like these intention-revealing variable names!

File: readme.md [hunk 3]
```diff
@@ -30,35 +30,32 @@ git clone https://github.com/karen-olson/serververse
...
 
-```
-./gradlew test
-```
-
-### Launching the program
+### Ending the program
 
-To run the program internally in IntelliJ, select the Run button or use ctrl + shift + R.
+To stop the server, enter CTRL+C in the terminal.
```
Comment (damonkelley) [inline readme.md:49]: **praise**: 🎉 

I appreciate it when instructions are independent of an IDE!

File: bin/acceptance-test (hunk 1 above)
Comment (karen-olson) [reply bin/acceptance-test]: Forgot to remove some of the echo statements, thanks for catching that!

**Pull request merged**